src/main/scala/org/constellation/ConstellationExecutionContext.scala

```diff
@@ -66,4 +69,10 @@ object ConstellationExecutionContext {
       new DefaultThreadFactory("callbacks")
     )
   )
+
+  def createSemaphore[F[_]: Concurrent](): Semaphore[F] = {
+    implicit val cs: ContextShift[IO] = IO.contextShift(ConstellationExecutionContext.bounded)
```

Comment:
Isn't it the same object? Why you need `ConstellationExecutionContext` here?

Reply:
fixed